native/handlers.go

```diff
@@ -15,14 +16,39 @@ func (i *Indexer) HandleNewBlock(ctx context.Context, blk *tmtypes.Block) error
 	return i.HandleBlock(ctx, blk)
 }
 
-// HandleBlock handles the receive of an block from the chain.
+// HandleBlock handles the receive of a block from the chain.
 func (i *Indexer) HandleBlock(ctx context.Context, blk *tmtypes.Block) error {
 	// light block
 	lb, err := i.b.LightProvider().LightBlock(ctx, blk.Header.Height)
 	if err != nil {
 		return err
 	}
 	i.logger.Info().Int64("light block", lb.SignedHeader.Header.Height).Msg("Light Block ")
+
+	signerAccount, err := CreateSigner(os.Getenv("SIGNER_ACCOUNT_MNEMONIC"))
+	if err != nil {
+		i.logger.Err(err).Msg("Error creating signer:")
+		return err
+	}
+
+	i.logger.Info().Str("signer address", signerAccount.Address).Msg("Light Block ")
+
+	gasObj := os.Getenv("GAS_ADDRESS")
+	i.logger.Info().Str("gas address", gasObj).Msg("Light Block ")
+
+	rsp, err := callMoveFunction(ctx, i.cli, signerAccount.Address, gasObj, lb)
```

Comment:
Why we have 2 function callMoveFunction and executeTransaction?

Reply:
you want me to have this in one function?